src/annotator/guest.ts

```diff
@@ -653,6 +654,37 @@ export class Guest
       this._integration.navigateToSegment?.(annotation),
     );
 
+    this._sidebarRPC.on(
+      'renderThumbnail',
+      (
+        tag: string,
+        options: RenderToBitmapOptions,
+        callback: (err: string | null, bitmap: ImageBitmap | null) => void,
```

Comment:
I know this signature has been widely used in the time before Promises took over, but maybe we could use a more strict signature with a discriminator type, which does not, from a type system point of view, allow for both an error and a bitmap to be provided, or both to be `null`, and therefore requires less checks when the result is evaluated.

```ts
type RenderThumbnailResult =
  | { type: 'success'; bitmap: ImageBitmap }
  | { type: 'error'; error: Error }

callback: (result: RenderThumbnailResult) => void,
```

Reply:
I agree. I followed this older pattern because that is what had been used for RPC callbacks in the past. I'll post a change along these lines as a follow-up.